app/component/itinerary/navigator/hooks/useRealtimeLegs.js

```diff
@@ -119,12 +119,11 @@ function getLegsOfInterest(initialLegs, time, previousFinishedLeg) {
     return acc;
   }, []);
 
-  const nextLegIdx = legs.findIndex(({ start }) => legTime(start) > time);
   let currentLeg = legs.find(
     ({ start, end }) => legTime(start) <= time && legTime(end) >= time,
   );
   let previousLeg = legs.findLast(({ end }) => legTime(end) < time);
-  let nextLeg = legs[nextLegIdx];
+  const nextLeg = legs.find(({ start }) => legTime(start) > time);
 
```

Comment:
This line seems now obsolete as the returned `nextLeg` property is calculated at the end by using `currentLeg.end` (or `time`)

Reply:
That version of nextLeg (one which can be waitleg) is used when flapping occurs.